controller/src/beerocks/master/tasks/dynamic_channel_selection_task.cpp

```diff
@@ -31,14 +31,151 @@ dynamic_channel_selection_task::dynamic_channel_selection_task(db &database_,
+        request->scan_params().dwell_time_ms     = dwell_time_msec;
+        request->scan_params().channel_pool_size = curr_channel_pool.size();
+        std::copy(curr_channel_pool.begin(), curr_channel_pool.end(),
+                  request->scan_params().channel_pool);
+
+        // get the parent node to send the CMDU to the agent
+        auto radio_mac_str = beerocks::net::network_utils::mac_to_string(m_radio_mac);
+        auto ire           = database.get_node_parent_ire(radio_mac_str);
+        son_actions::send_cmdu_to_agent(ire, cmdu_tx, database, radio_mac_str);
+
+        set_events_timeout(SCAN_TRIGGERED_WAIT_TIME_MSEC);
+        dcs_wait_for_event(eEvent::SCAN_TRIGGERED);
+
+        fsm_move_state(eState::WAIT_FOR_SCAN_TRIGGERED);
+        break;
+    }
+    case eState::WAIT_FOR_SCAN_TRIGGERED: {
+        TASK_LOG(ERROR) << "This should not happen!";
+
+        m_last_scan_error_code =
+            beerocks::eChannelScanErrCode::CHANNEL_SCAN_TRIGGERED_EVENT_TIMEOUT;
+
+        fsm_move_state(eState::ABORT_SCAN);
+        break;
+    }
+    case eState::WAIT_FOR_RESULTS_READY: {
+        TASK_LOG(ERROR) << "This should not happen!";
+
+        m_last_scan_error_code =
+            beerocks::eChannelScanErrCode::CHANNEL_SCAN_RESULTS_READY_EVENT_TIMEOUT;
+
+        fsm_move_state(eState::ABORT_SCAN);
+        break;
+    }
+    case eState::WAIT_FOR_RESULTS_DUMP: {
+        TASK_LOG(ERROR) << "This should not happen!";
+
+        m_last_scan_error_code =
+            beerocks::eChannelScanErrCode::CHANNEL_SCAN_RESULTS_DUMP_EVENT_TIMEOUT;
+
+        fsm_move_state(eState::ABORT_SCAN);
+        break;
+    }
+    case eState::SCAN_DONE: {
+        LOG(TRACE) << "SCAN_DONE";
+
+        //update next continuous scan time
+        if (!m_is_single_scan) {
+            auto interval =
+                std::chrono::seconds(database.get_channel_scan_interval_sec(m_radio_mac));
+            m_next_scan_timestamp_interval = m_last_scan_try_timestamp + interval;
+        }
+
+        m_last_scan_error_code = beerocks::eChannelScanErrCode::CHANNEL_SCAN_SUCCESS;
+        database.set_channel_scan_results_status(m_radio_mac, m_last_scan_error_code,
+                                                 m_is_single_scan);
+
+        database.set_channel_scan_in_progress(m_radio_mac, false, m_is_single_scan);
+        m_is_single_scan = false;
+        fsm_move_state(eState::IDLE);
+        break;
+    }
+    case eState::ABORT_SCAN: {
+        LOG(ERROR) << "aborting scan for mac=" << m_radio_mac << ", last_scan_timestamp is not set";
+
+        database.set_channel_scan_results_status(m_radio_mac, m_last_scan_error_code,
+                                                 m_is_single_scan);
+        database.set_channel_scan_in_progress(m_radio_mac, false, m_is_single_scan);
+        m_is_single_scan = false;
+
+        fsm_move_state(eState::IDLE);
+        work(); // Skip pause, move to Idle immediately
```

Comment:
I don't understand this - we shouldn't call work() directly, should we? So why is it done here?

Reply:
proposed by @moran in order not to wait for next after_select to reach Idle.

Reply:
But why? if it is for "performance" - please remove it. Readability is more important.

Reply:
yes performance.
removed